automation/test-execution/ansible/scripts/log_to_mlflow.py

```
@@ -0,0 +1,636 @@
+
+        # Check if this test_run_id was already logged (deduplication)
+        existing_runs = mlflow.search_runs(
+            experiment_ids=[experiment.experiment_id],
+            filter_string=f"params.test_run_id = '{metadata.get('test_run_id', 'unknown')}'",
+            max_results=1
+        )
+
+        if not existing_runs.empty:
+            existing_run_id = existing_runs.iloc[0]['run_id']
+            existing_run_name = existing_runs.iloc[0]['tags.mlflow.runName']
+            print(f"⚠️  Test already logged to MLflow")
```

Ruff (0.15.12)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)